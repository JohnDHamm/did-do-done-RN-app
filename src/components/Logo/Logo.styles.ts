import styled from 'styled-components/native';

export const StyledImage = styled.Image<{
  color: string;
  width: number;
  height: number;
}>`
  width: ${(props) => props.width + 'px'};
  height: ${(props) => props.height + 'px'};
  tint-color: ${(props) => props.color};
`;
