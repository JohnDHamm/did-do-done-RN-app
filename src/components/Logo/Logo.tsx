import React from 'react';
import { StyledImage } from './Logo.styles';
import IMAGES from '../../../assets/images';

interface Props {
  color: string;
  width: number;
}

const Logo: React.FC<Props> = ({ color, width }) => {
  return (
    <StyledImage
      source={IMAGES.LOGO}
      color={color}
      width={width}
      height={width / IMAGES.LOGO_ASPECT}
    />
  );
};

export default Logo;
