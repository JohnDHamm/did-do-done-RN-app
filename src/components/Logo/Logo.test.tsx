import React from 'react';
import { cleanup, render } from 'react-native-testing-library';
import Logo from './Logo';

describe('Logo', () => {
  afterEach(cleanup);

  it('renders correctly', () => {
    const { toJSON } = render(<Logo color="purple" width={100} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
