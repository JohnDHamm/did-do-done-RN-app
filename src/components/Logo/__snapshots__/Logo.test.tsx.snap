// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<Image
  color="purple"
  height={12.5}
  source={1}
  style={
    Array [
      Object {
        "height": 12.5,
        "tintColor": "purple",
        "width": 100,
      },
    ]
  }
  width={100}
/>
`;
